AutoCollection/AzureFunctionsHook.ts

```diff
@@ -0,0 +1,76 @@
+import { Context, HttpRequest } from "../Library/Functions";
+import Logging = require("../Library/Logging");
+import TelemetryClient = require("../Library/TelemetryClient");
+import { CorrelationContextManager } from "./CorrelationContextManager";
+
+
+/** Node.js Azure Functions handle incoming HTTP requests before Application Insights SDK is available,
+ * this code generate incoming request telemetry and generate correlation context to be used 
+ * by outgoing requests and other telemetry, we rely on hooks provided by Azure Functions
+*/
+export class AutoCollectAzureFunctions {
+    private _client: TelemetryClient;
+    private _functionsCoreModule: any;
+    private _preInvocationHook: any;
+
+    constructor(client: TelemetryClient) {
+        this._client = client;
+        try {
+            this._functionsCoreModule = require('@azure/functions-core');
+        }
+        catch (error) {
+            Logging.info("AutoCollectAzureFunctions failed to load, not running in Azure Functions");
+        }
+    }
+
+    public enable(isEnabled: boolean) {
```

Comment:
Do you want to add a check to block "double" enabling?

i.e. someone calls enable(true) more than once.

Reply:
Should be handled already by _addPreInvocationHook